File: Python/carrinhoCompras.py
carrinho = []

while True:

    produto = input("Nome do produto: ")
    preco = float(input("Preço do produto: "))

    carrinho.append({"nome": produto, "preco": preco})

    continuar = input("Adicionar outro produto? (s/n): ")

    if continuar.lower() != "s":
        break

total = 0

print("\nProdutos no carrinho:")

for item in carrinho:

    print(item["nome"], "- R$", item["preco"])
    total += item["preco"]

print("\nTotal da compra: R$", total)
print("Quantidade de produtos:", len(carrinho))

File: Java/CarrinhoCompras.java
import java.util.ArrayList;
import java.util.Scanner;

class Produto{
    String nome;
    double preco;

    Produto(String nome, double preco){
        this.nome = nome;
        this.preco = preco;
    }
}

public class CarrinhoCompras {

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);
        ArrayList<Produto> carrinho = new ArrayList<>();

        while(true){

            System.out.print("Nome do produto: ");
            String nome = sc.nextLine();

            System.out.print("Preço do produto: ");
            double preco = sc.nextDouble();
            sc.nextLine();

            carrinho.add(new Produto(nome, preco));

            System.out.print("Adicionar outro produto? (s/n): ");
            String resp = sc.nextLine();

            if(!resp.equalsIgnoreCase("s")){
                break;
            }
        }

        double total = 0;

        System.out.println("\nProdutos no carrinho:");

        for(Produto p : carrinho){

            System.out.println(p.nome + " - R$ " + p.preco);

            total += p.preco;
        }

        System.out.println("\nTotal da compra: R$ " + total);
        System.out.println("Quantidade de produtos: " + carrinho.size());

        sc.close();
    }
}